PipelinesTasks/PRMetrics/buildAndReleaseTask/invokers/azureReposInvoker.ts

```diff
@@ -0,0 +1,219 @@
+    private taskLibWrapper: TaskLibWrapper;
+    private devOpsApiWrapper: DevOpsApiWrapper;
+    private gitApi: IGitApi | undefined;
+
+    private baseUri = `${process.env["SYSTEM_TEAMFOUNDATIONCOLLECTIONURI"]}`;
+    private project = process.env["SYSTEM_TEAMPROJECT"];
+    private repositoryId = process.env["BUILD_REPOSITORY_ID"] as string;
+    private pullRequestId = process.env["SYSTEM_PULLREQUEST_PULLREQUESTID"] ? parseInt(process.env["SYSTEM_PULLREQUEST_PULLREQUESTID"]) : 0;
+    private azurePAT = process.env["SYSTEM_ACCESSTOKEN"];
+
+    /**
+      * Initializes a new instance of the AzureReposInvoker class.
+      * @param devOpsApiWrapper The wrapper around the Azure Devops Api Task Lib.
+      * @param taskLibWrapper The wrapper around the Azure Pipelines Task Lib.
+      */
+    public constructor(devOpsApiWrapper: DevOpsApiWrapper, taskLibWrapper: TaskLibWrapper) {
+        this.devOpsApiWrapper = devOpsApiWrapper;
+        this.taskLibWrapper = taskLibWrapper;
+    }
+
+    /**
+      * Gets the pull request from the devops api.
+      */
+    public getDetails = async (): Promise<GitPullRequest> => {
+        this.taskLibWrapper.debug('* AzureReposInvoker.getDetails()');
+
+        const gitApi = await this.openConnection();
+        return await gitApi.getPullRequestById(this.pullRequestId, this.project);
+    }
+
+    /**
+      * Gets the pull request iterations from the devops api.
+      */
+    public getIterations = async (): Promise<GitPullRequestIteration[]> => {
+        this.taskLibWrapper.debug('* AzureReposInvoker.getIterations()');
+
+        const gitApi = await this.openConnection();
+        return await gitApi.getPullRequestIterations(this.repositoryId, this.pullRequestId, this.project);
+    }
+
+    /**
+      * Gets the pull request comment threads from the devops api.
+      */
+    public getCommentThreads = async (): Promise<GitPullRequestCommentThread[]> => {
+        this.taskLibWrapper.debug('* AzureReposInvoker.getCommentThreads()');
+
+        const gitApi = await this.openConnection();
+        return await gitApi.getThreads(this.repositoryId, this.pullRequestId, this.project);
+    }
+
+    /**
+      * Gets the pull request comment thread from the devops api.
+      * @param commentThreadId Comment thread id to be used to retrieve comment thread.
+      */
+    public getCommentThread = async (commentThreadId: number): Promise<GitPullRequestCommentThread> => {
+        this.taskLibWrapper.debug('* AzureReposInvoker.getCommentThread()');
+
+        const gitApi = await this.openConnection();
+        return await gitApi.getPullRequestThread(this.repositoryId, this.pullRequestId, commentThreadId, this.project);
+    }
+
+    /**
+      * Updates the description and title of the pull request.
+      * @param description New pull request description.
+      * @param title New pull request title.
+      */
+    public setDetails = async (description: string, title: string): Promise<void> => {
```

Comment:
Can you make these nullable as well? I'm going to send null now if we don't want to update